### PR TITLE
[Oy2-19466] Create Header Component

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Creates the basic `Header` and other associated components.

## Type of Request

feature

## Approach

Uses the USWDS standard header as a foundation. There is some additional styling that has been added for the purpose of matching the Figma design. Additionally, created an `ActionsMenu` component to be used as a part of the header. The `Header` takes a child component that will be rendered in the upper right on desktop, and tucked inside the side menu on mobile.

I did not create a Storybook story for the `ActionsMenu`. This is not intended to be used outside the context of the `Header`. The `Header` story shows this implementation. The code for both components lives in separate files in the same directory.

#### Pull Request Creator Checklist

- [X] It is documented what type of request this is.
- [X] There is a detailed description for the purpose of this PR.
- [X] This PR is in line with the intention of this Project

### CODE REVIEW

File: src/components/Header/Header.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  },
+];
+
+export default {
+  title: "USWDS/Base/Header",
+  component: Header,
+  args: {
+    logoProps: {
+      altText: "CMS.gov Project",
+      // source: "../../assets/img/logos/cms_logo.png",
```
Comment (jdinh8124) [inline src/components/Header/Header.stories.tsx:53]: Might need cleanup? I see what you're trying to show though, the alt text would show in a typography component.

File: .storybook/main.js [hunk 2]
```diff
@@ -12,4 +14,13 @@ module.exports = {
...
   staticDirs: ['../src', '../src/assets/img'],
+
+  webpackFinal: async (config) => {
+    config.resolve.modules = [
+      ...(config.resolve.modules || []),
+      path.resolve(__dirname, "../src"),
+    ];
+
+    return config;
+  },
```
[the remark this answers is not in the document]
Comment (charmcitygavin) [reply .storybook/main.js:25]: I'm also curious to learn more about this. It looks cool, whatever it's doing 😂

File: src/components/Header/ActionsMenu.scss [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    cursor: pointer;
+    display: flex;
+    flex-direction: row;
+    font-weight: bold;
+    position: relative;
+  }
+
+  .user-actions-menu {
+    display: none;
+    background-color: #0071bc;
```
Comment (charmcitygavin) [inline src/components/Header/ActionsMenu.scss:35]: FYI, in my current branch, I've created a new SCSS file that captures all of these MACBIS colors as reusable variables. Figured that could help us stay consistent and make it easy to update any colors if design wants to tweak something. 😅 I'll be happy to walk you through it at stand-up

File: src/components/Header/ActionsMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { Icon, IconChoice } from "components/Icon/Icon";
+
+interface Link {
+  href?: string;
+  iconName?: IconChoice;
+  onClick?: () => any;
+  text: string;
+}
+
```
Comment (jdinh8124) [inline src/components/Header/ActionsMenu.tsx:10]: target?:

**Pull request merged**